### PR TITLE
Move test harness to inline c

### PR DESCRIPTION
**Pull request by mineo333**
This change hopes to address some of the concerns described in https://github.com/AFLplusplus/LibAFL/issues/1799. 

This PR moves the Frida ASAN test from an external cpp file to inline c. This is done by having the `#inline_c_rs CFLAGS: "-shared"` flag at the top which forces it to compile as a shared library. The shared library is then loaded via libloading. 

Note that this may break the tests in https://github.com/AFLplusplus/LibAFL/pull/1607. However, if needed, more flags can be introduced.

### CODE REVIEW

**Comment by s1341** [comment]
Thanks!

I'd like to see:
1. That we can build and run tests on windows too using `cargo xwin`.
2. a macro for defining a test case, which generates the boiler plate necessary, when provided with a C++ block and some expectations.

**Comment by mineo333** [comment]
Due to the substantial changes needed to inline-c, a fork has been created here: https://github.com/mineo333/libafl-inline-c-rs

Support for compiling as a shared library and cross-compilation is being added.

**Comment by domenukk** [comment]
What's the benefit over simply having a c file lying around? Isn't this way more confusing for new developers joining the project? Also, we won't have clang format nor code completion for inline C, right?

File: libafl_frida/Cargo.toml [hunk 1]
```diff
@@ -86,3 +86,4 @@ yaxpeax-arch = "0.2.7"
 serial_test = { version = "2", default-features = false, features = ["logging"] }
 clap = {version = "4.0", features = ["derive"]}
 libloading = "0.7"
+inline-c = {git = 'https://github.com/mineo333/inline-c-rs' }
```
Comment (domenukk) [inline libafl_frida/Cargo.toml:89]: Please either try to get the changes upstreamed or publish it as new library on crates.io

**Comment by mineo333** [comment]
> What's the benefit over simply having a c file lying around? Isn't this way more confusing for new developers joining the project? Also, we won't have clang format nor code completion for inline C, right?

I think it could be easier in the long term mainly because the cc stuff would be abstracted out via the inline_c_rs options at the top. We also wouldn't have a growing build script every time we need to have a test. However, there definitely would be a small learning curve when using this. 

Also, clang format nor code completion would be available with this.

**Comment by s1341** [comment]
I agree... The C/C++ snippets are parts of the test fixture. It's very annoying when the test fixure is spread over mulitiple files/localtions. 

With this, you have the whole test self-contained in a single location.

**Comment by s1341** [comment]
Maybe break it up into individual tests?

**Comment by s1341** [comment]
Let's see how we can refine the test macro as we build out some tests...

**Comment by domenukk** [comment]
What's the status?

**Comment by mineo333** [comment]
Right now, in terms of splitting the tests, there's a weird problem where because the frida hooks are not deinit'd, it is effectively impossible to create multiple frida helpers. The ideal solution for that is what @s1341 implemented in https://github.com/AFLplusplus/LibAFL/pull/1607/ with callouts, so I'm going to see what I can do with that.

**Pull request closed without merging**